app/models/claim.rb

```diff
@@ -30,17 +30,17 @@ class Claim < ApplicationRecord
   belongs_to :provider
   belongs_to :created_by, polymorphic: true
 
-  has_many :mentor_trainings
+  has_many :mentor_trainings, dependent: :destroy
   has_many :mentors, through: :mentor_trainings
 
   validates :status, presence: true
   validates :reference, uniqueness: { case_sensitive: false }, allow_nil: true
 
-  scope :not_internal, -> { where.not(status: :internal) }
+  scope :not_internal_or_discarded, -> { where.not(status: %i[internal discarded]) }
   scope :order_created_at_desc, -> { order(created_at: :desc) }
 
   enum :status,
-       { internal: "internal", draft: "draft", submitted: "submitted" },
+       { internal: "internal", draft: "draft", submitted: "submitted", discarded: "discarded" },
```

Comment:
Why aren't you using the `discard` gem for making a record discarded? We already have this functionality for `Users`, and should keep the discarding mechanism consistent throughout the service/application

Reply:
A key feature of using `discarded` is that, should you need to `undiscard` the record you haven't lost it's original state.